### PR TITLE
DOC: improve docstrings of interpolate.NearestNDInterpolator.__call__ & LinearNDInterpolator.__call__

### LINKED ISSUE #12378: interpolate.NearestNDInterpolator.__call__ & LinearNDInterpolator.__call__ documentation
Documentation states that __call__ method takes a single argument xi as an array of points with shape=(..., ndim), but the __call__ method also accepts ndim broadcastible arrays as an argument. This capability stems from code line:

`xi = _ndim_coords_from_arrays(args, ndim=self.points.shape[1])`

This option for evaluation should be included in the documentation.

I suggest the documentation be changed to something like:

> Parameters:
> -------------
>     x1, x2, ... xndim: array_like of float
>          Points where to interpolate data at.
> 
>          x1, x2, ... xndim can be array_like of float with broadcastible shape.
> 
>          or x1 can be array_like of float with shape (..., ndim)

The follwing code demonstrates the two modes of calling working as expected.

```
import numpy as np
import scipy
from scipy.interpolate import NearestNDInterpolator, LinearNDInterpolator

print('numpy version = ', np.version.full_version)
print('scipy version = ', scipy.version.full_version)


for itype in (NearestNDInterpolator, LinearNDInterpolator):
    shape = (4, 5, 6, 7)
    ndims = len(shape)
    x_0 = [np.arange(0, n, dtype=float) for n in shape]
    pts = np.moveaxis(
        np.meshgrid(*x_0, indexing='ij'), 0, -1
        ).reshape(-1, ndims)
    values = np.sum(pts, axis=1)
    interp = itype(pts, values)
    print(40*'-')
    print(itype.__name__)
    print(40*'-')
    print('interp((3, 3, 3, 3)) = ', interp((3, 3, 3, 3)), '\n')
    print('interp(3, 3, 3, 3) = ', interp(3, 3, 3, 3), '\n')
    print(
        'interp([[2], [3]], [[0, 1, 2]], 3, 3) = ',
        interp([[2], [3]], [[0, 1, 2]], 3, 3), '\n'
        )
```
```

numpy version =  1.18.1
scipy version =  1.4.1
----------------------------------------
NearestNDInterpolator
----------------------------------------
interp((3, 3, 3, 3)) =  12.0 

interp(3, 3, 3, 3) =  12.0 

interp([[2], [3]], [[0, 1, 2]], 3, 3) =  [[ 8.  9. 10.]
 [ 9. 10. 11.]] 

----------------------------------------
LinearNDInterpolator
----------------------------------------
interp((3, 3, 3, 3)) =  12.0 

interp(3, 3, 3, 3) =  12.0 

interp([[2], [3]], [[0, 1, 2]], 3, 3) =  [[ 8.  9. 10.]
 [ 9. 10. 11.]] 

```

### PR DESCRIPTION
**Pull request by AtsushiSakai**
#### Reference issue
fix #12378 

#### What does this implement/fix?
I improved docstrings `interpolate.NearestNDInterpolator.__call__` & `LinearNDInterpolator.__call__`.

Because, these functions also accepts ndim broadcastible arrays as an argument.

### CODE REVIEW

**Comment by rgommers** [comment]
Hmm, I'm not sure that helps users. Let me comment on the issue.

**Comment by AtsushiSakai** [comment]
@rgommers Hi. Thank you for your comment. I added a test based on your comment. PTAL.

File: scipy/interpolate/interpnd.pyx [hunk 1]
```diff
@@ -133,8 +133,10 @@ class NDInterpolatorBase(object):
 
         Parameters
         ----------
-        xi : ndarray of float, shape (..., ndim)
+        x1, x2, ... xn: array_like of float
             Points where to interpolate data at.
+            x1, x2, ... xn can be array_like of float with broadcastable shape.
+            or x1 can be array_like of float with shape (..., ndim)
```
Comment (rgommers) [inline scipy/interpolate/interpnd.pyx:139]: two minor things:
- `array_like` -> `array-like`
- would be nice to have double backticks around `(..., ndim)`

Comment (rgommers) [reply scipy/interpolate/interpnd.pyx:139]: same applies to the other docstring further down

File: scipy/interpolate/tests/test_interpolate.py [hunk 2]
```diff
@@ -2581,6 +2582,32 @@ def test_fillvalue_type(self):
...
+        y = np.random.random(10)
+        z = np.hypot(x, y)
+
+        # x-y grid for interpolation
+        X = np.linspace(min(x), max(x))
+        Y = np.linspace(min(y), max(y))
+        X, Y = np.meshgrid(X, Y)
+        XY = np.vstack((X.ravel(), Y.ravel())).T
+
+        for itype in (NearestNDInterpolator, LinearNDInterpolator,
```
Comment (rgommers) [inline scipy/interpolate/tests/test_interpolate.py:2598]: nit: could you rename `itype` to `interpolator` or some other more descriptive name? I read the `itype(list(zip` line first and confusingly read it as "integer type".

File: scipy/interpolate/tests/test_interpolate.py [hunk 3]
```diff
@@ -2581,6 +2582,32 @@ def test_fillvalue_type(self):
...
+                      CloughTocher2DInterpolator):
+            interp = itype(list(zip(x, y)), z)
+            # single array input
+            interp_points0 = interp(XY)
+            # broadcastable input
+            interp_points1 = interp((X, Y))
```
Comment (rgommers) [inline scipy/interpolate/tests/test_interpolate.py:2604]: This is a tuple of 1-D array input. I think what should be tested here is two positional args, both 1-D arrays. So `interp(X, Y)`. And same for the line below.

**Review by rgommers — COMMENTED** [review]
Thanks @AtsushiSakai, looks about right. A few small comments left.

File: scipy/interpolate/interpnd.pyx (hunk 1 above)
Comment (AtsushiSakai) [reply scipy/interpolate/interpnd.pyx:139]: Done.

File: scipy/interpolate/tests/test_interpolate.py (hunk 2 above)
Comment (AtsushiSakai) [reply scipy/interpolate/tests/test_interpolate.py]: I agree. I fixed it.

File: scipy/interpolate/tests/test_interpolate.py (hunk 3 above)
Comment (AtsushiSakai) [reply scipy/interpolate/tests/test_interpolate.py:2604]: Oh, OK. I fixed it.

**Comment by rgommers** [comment]
LGTM now, merged. Thanks @AtsushiSakai!

**Pull request merged**